test/specs/routing.test.js

```diff
@@ -580,6 +580,16 @@ module.exports = fetchMock => {
 					expect(fm.calls(true).length).to.equal(1);
 				});
 
+				it('should match when using Request', async () => {
+					fm.mock('http://it.at.there/', 200, { body: { foo: 'bar' } }).catch();
+
+					await fm.fetchHandler(new fm.config.Request('http://it.at.there/', {
```

Comment:
About your concerns of introducing async handling: It just feels natural using `async` / `await` since we are dealing with network requests that happens to be async by default.

Reply:
Yeah, I think the benefit of supporting ky outweighs the concern. It has to be a major release though so I will hold off while I consider whether there are other ways the API can be improved at the same time e.g. https://github.com/wheresrhys/fetch-mock/issues/489

Reply:
I'm looking at this again. Another thing that breaks when introducing this early asynchronicity is it is not possible to check synchronously that an expected call has been made

i.e. at the moment the following works:

```
myMethod() // which fetches 'http://alan.com'
expect(fetchMock.called('http://alan.com'))
```

but it will ned to be updated to 

```
await myMethod() // which fetches 'http://alan.com'
expect(fetchMock.called('http://alan.com'))
```
or

```
myMethod() // which fetches 'http://alan.com'
await fetchMock.flush()
expect(fetchMock.called('http://alan.com'))
```

While these may seem minor, it's potentially imposing lots of annoying little test rewrites on users. I'm not enjoying doing this for fetch-mock's own test suite, and don't particularly want to cause others that pain.

Still going to try to solve this in a nice way, but needs quite a bit of thought